store/src/java/com/zimbra/cs/service/mail/SaveDraft.java

```diff
@@ -101,8 +101,11 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         Mailbox delegatorMbox = null;
         if (mailAddress != null) {
             delegatorAccount = prov.get(AccountBy.name, mailAddress, zsc.getAuthToken());
-            String mRequestedAccountId = delegatorAccount.getId();
-            delegatorMbox = MailboxManager.getInstance().getMailboxByAccountId(mRequestedAccountId, true);
+            String mRequestedAccountId;
+            if (delegatorAccount != null) {
+                mRequestedAccountId = delegatorAccount.getId();
```

Comment:
can we use `delegatorMbox = MailboxManager.getInstance().getMailboxByAccountId(delegatorAccount.getId(), true);`
and remove need of `String mRequestedAccountId;`